web-admin/src/features/dashboards/DashboardAdminStateProvider.svelte

```diff
@@ -1,74 +1,33 @@
 <script lang="ts">
   import { page } from "$app/stores";
   import { useQueryClient } from "@rilldata/svelte-query";
-  import { getBookmarks } from "@rilldata/web-admin/features/bookmarks/selectors";
-  import {
-    createTimeRangeSummary,
-    useMetricsView,
-    useModelHasTimeSeries,
-  } from "@rilldata/web-common/features/dashboards/selectors/index";
+  import { getHomeBookmarkData } from "@rilldata/web-admin/features/bookmarks/selectors";
   import { getStateManagers } from "@rilldata/web-common/features/dashboards/state-managers/state-managers";
-  import { metricsExplorerStore } from "@rilldata/web-common/features/dashboards/stores/dashboard-stores";
-  import { hasPersistentDashboardData } from "@rilldata/web-common/features/dashboards/stores/persistent-dashboard-state";
-  import { syncDashboardState } from "@rilldata/web-common/features/dashboards/stores/syncDashboardState";
+  import { createDashboardStateSync } from "@rilldata/web-common/features/dashboards/stores/syncDashboardState";
   import { initLocalUserPreferenceStore } from "@rilldata/web-common/features/dashboards/user-preferences";
-  import { createQueryServiceMetricsViewSchema } from "@rilldata/web-common/runtime-client";
   import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
   import Spinner from "@rilldata/web-common/features/entity-management/Spinner.svelte";
   import { EntityStatus } from "@rilldata/web-common/features/entity-management/types";
 
   export let metricViewName: string;
 
   $: initLocalUserPreferenceStore(metricViewName);
-  const stateManagers = getStateManagers();
-  const metricsView = useMetricsView(stateManagers);
-  const hasTimeSeries = useModelHasTimeSeries(stateManagers);
-  const timeRangeQuery = createTimeRangeSummary(stateManagers);
-  const metricsViewSchema = createQueryServiceMetricsViewSchema(
-    $runtime.instanceId,
-    metricViewName,
-  );
-
   const queryClient = useQueryClient();
-  $: bookmarks = getBookmarks(
+  const homeBookmark = getHomeBookmarkData(
```

Comment:
Looks like this should be reactive?

Reply:
Since none of those can change during a dashboard being open it can be non reactive. Also we have a key on `metricsViewName` so this will always have the correct store.